hknweb/academics/models/logistics.py

```diff
@@ -0,0 +1,30 @@
+from django.db import models
+
+from .base_models import AcademicEntity
+
+
+class Course(AcademicEntity):
+    pass
+
+
+class Department(AcademicEntity):
+    name = models.TextField(max_length=200)
+    abbr = models.TextField(max_length=100)
+
+
+class Instructor(AcademicEntity):
+    instructor_id = models.TextField(max_length=500)
+
+
+class Semester(AcademicEntity):
+    SPRING = 'Sp'
+    SUMMER = 'Su'
+    FALL = 'Fa'
+    YEAR_SECTION_CHOICES = [
+        (SPRING, 'Spring'),
+        (SUMMER, 'Summer'),
+        (FALL, 'Fall'),
+    ]
+
+    year = models.TextField(max_length=100)

```

Comment:
A string year? Bold move. (Might work better for formatting but may cause surprise TypeErrors on comparisons to integer years.)

Reply:
The formatting was exactly my consideration; I'll take a look and see what the hknweb precedent is. If it's integer years, then we'll switch over to integer years.

Reply:
[`alumni/models.py/Alumnus.year`](https://github.com/compserv/hknweb/blob/57d815b738f7449732d830ea4d44f6304bccbfca/hknweb/alumni/models.py#L40-L43) uses an IntegerField with validation. [`tutoring/models.py/Slot.day`](https://github.com/compserv/hknweb/blob/57d815b738f7449732d830ea4d44f6304bccbfca/hknweb/tutoring/models.py#L43) also uses IntegerField. IntegerField with validation it is!

Reply:
Leaving out any even slightly more complicated validation logic, leaving in a todo with a reference to the alumni example.